src/Wig.Zstd/Program.cs

```diff
@@ -29,36 +35,143 @@ public sealed class DefaultCommand : AsyncCommand<DefaultCommand.Settings>
+            [CommandOption("-l|--level")]
+            [Description("Compression level (1-22)")]
+            public int CompressionLevel { get; set; }
+
+            [CommandOption("-s|--subfolder")]
+            [Description("Subfolder name")]
+            public string Subfolder{ get; set; }
         }
 
         public DefaultCommand(IAnsiConsole console)
         {
             _console = console;
         }
 
+        public async Task WriteFileAsync(byte[] data, string path, string subfolder, string extension)
+        {
+            var fileName = Path.GetFileName(path);
+            var currentDirectory = Path.GetDirectoryName(path);
+
+            var directory = Directory.CreateDirectory($"{currentDirectory}/{subfolder}");
+
+            await using (FileStream fstream = new FileStream($"{directory}/{fileName}{extension}", FileMode.OpenOrCreate))
+            {
+                await fstream.WriteAsync(data, 0, data.Length);
+            }
+        }
+
+        public async Task CompressingAsync(string path, int compressionLevel, bool overwrite, string subfolder)
+        {
+            using var options = new CompressionOptions(compressionLevel);
+            using var compressor = new Compressor(options);
+            FileAttributes attr = File.GetAttributes(path);
+            if (attr.HasFlag(FileAttributes.Directory))
+            {
+                string[] filePaths = Directory.GetFiles(path);
+
+                foreach (var filePath in filePaths)
+                {
+                    byte[] data = await File.ReadAllBytesAsync(filePath);
+
+                    var compData = compressor.Wrap(data);
+
+                    if (File.Exists($"{filePath}.zs") && !overwrite)
+                    {
+                        AnsiConsole.WriteLine("A compressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                        return;
+                    }
+
+                    await WriteFileAsync(compData, path, subfolder, "zs");
+                }
+            }
+
+            byte[] sourceData = await File.ReadAllBytesAsync(path);
+
+            var compressedData = compressor.Wrap(sourceData);
+
+            if (File.Exists($"{path}.zs") && !overwrite)
+            {
+                AnsiConsole.WriteLine("A compressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                return;
+            }
+
+            await WriteFileAsync(compressedData, path, subfolder, ".zs");
+        }
+
+        public async Task DecompressingAsync(string path, bool overwrite, string subfolder)
+        {
+            using var decompressor = new Decompressor();
+
+            FileAttributes attr = File.GetAttributes(path);
+            if (attr.HasFlag(FileAttributes.Directory))
+            {
+                string[] filePaths = Directory.GetFiles(path);
+                foreach (var filePath in filePaths)
+                {
```

Comment:
it can be moved into separate method

as you can see, this code is duplicated
```
if (filePath.Contains(".zs"))
{
  byte[] compressedData = await File.ReadAllBytesAsync($"{filePath}");
  var decompressedData = decompressor.Unwrap(compressedData);
  var unpackingPath = Path.ChangeExtension(filePath, "");

  if (File.Exists($"{path}") && !overwrite)
  {
    AnsiConsole.WriteLine("A decompressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
    return;
  }

  await WriteFileAsync(decompressedData, unpackingPath, subfolder, "");
}
```